src/execution/command_parser.c

```diff
@@ -113,24 +142,30 @@ static int _ModelRunCtx_SetParams(RedisModuleCtx *ctx, RedisModuleString **inkey
     }
 
     for (size_t i = 0; i < noutputs; i++) {
-        if (model->outputs)
+        if (model->outputs) {
             opname = model->outputs[i];
+        }
+        if (!VerifyKeyInThisShard(ctx, outkeys[i])) { // Relevant for enterprise cluster.
+            RAI_SetError(err, RAI_EMODELRUN,
+                         "ERR CROSSSLOT Keys in request don't hash to the same slot");
+            return REDISMODULE_ERR;
```

Comment:
move this to the actual parsing phase. fail as soon as possible when you can

Reply:
We can't because the parsing function is used from dag command as well (and in DAG we do not take the keys from key space...)
But `ModelRunCtx_SetParams` is called only from "pure" AI.MODELEXECUTE.